src/main/scala/org/broadinstitute/dsde/workbench/sam/api/StandardUserInfoDirectives.scala

```diff
@@ -13,56 +13,92 @@ import org.broadinstitute.dsde.workbench.sam.api.StandardUserInfoDirectives._
 import org.broadinstitute.dsde.workbench.sam.directory.DirectoryDAO
 import org.broadinstitute.dsde.workbench.sam.service.UserService._
 
-import scala.concurrent.{ExecutionContext, Future}
-import scala.util.Try
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success, Try}
 import pdi.jwt.{JwtOptions, JwtSprayJson}
 import spray.json._
 import java.time.Instant
 
-trait StandardUserInfoDirectives extends UserInfoDirectives {
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.sam.identityConcentrator.IdentityConcentratorService
+
+import scala.util.matching.Regex
+import DefaultJsonProtocol._
+import WorkbenchIdentityJsonSupport.identityConcentratorIdFormat
+
+trait StandardUserInfoDirectives extends UserInfoDirectives with LazyLogging {
   implicit val executionContext: ExecutionContext
+  val identityConcentratorService: Option[IdentityConcentratorService] = None
 
   def requireUserInfo: Directive1[UserInfo] =
-    (
-      optionalHeaderValueByName(accessTokenHeader) &
-        optionalHeaderValueByName(googleSubjectIdHeader) &
-        optionalHeaderValueByName(expiresInHeader) &
-        optionalHeaderValueByName(emailHeader) &
-        headerValueByName(authorizationHeader)
+    (optionalHeaderValueByName(accessTokenHeader) &
+      optionalHeaderValueByName(googleSubjectIdHeader) &
+      optionalHeaderValueByName(expiresInHeader) &
+      optionalHeaderValueByName(emailHeader) &
+      headerValueByName(authorizationHeader) &
+      provide(identityConcentratorService)
```

Comment:
using `provide` here so the existence or not of `identityConcentratorService` can fit into the pattern matching below